geode-core/src/main/java/org/apache/geode/internal/offheap/FreeListManager.java

```diff
@@ -518,10 +520,51 @@ boolean doDefragment(int chunkSize) {
     ma.getStats().setLargestFragment(largestFragment);
     ma.getStats().setFragments(tmp.size());
     ma.getStats().setFragmentation(getFragmentation());
+    ma.getStats().setFreedChunks(0);
 
     return result;
   }
 
+  public void updateNonRealTimeStats() {
+    ma.getStats().setLargestFragment(getLargestFragment());
+    ma.getStats().setFreedChunks(getFreedChunks());
+  }
+
+  public int getFreedChunks() {
+    int elementCountFromTinyFreeLists =
+        getElementCountFromTinyFreeLists();
+    int elementCountFromHugeFreeLists =
+        getElementCountFromHugeFreeLists();
+
+    return elementCountFromTinyFreeLists + elementCountFromHugeFreeLists;
+  }
+
+  private int getElementCountFromTinyFreeLists() {
+    int fragmentCount = 0;
+    for (int i = 0; i < tinyFreeLists.length(); i++) {
+      OffHeapStoredObjectAddressStack cl = tinyFreeLists.get(i);
+      if (cl != null) {
+        fragmentCount += cl.size();
+      }
+    }
+    return fragmentCount;
+  }
+
+  private int getElementCountFromHugeFreeLists() {
+    return hugeChunkSet.size();
+  }
+
+  private int getLargestFragment() {
+    int largestFreeSpaceFromFragments = 0;
```

Comment:
would "largestFragmentSize" be a better name? It should probably match the method name without the "get" prefix.

Reply:
@dschneider-pivotal It would probably be a better name. Should I also change the name of the stats and the methods to get them to `largestFragmentSize`, `setLargestFragmentSize()` and `getLargestFragmentSize()`?